pkg/migration/generate.go

```diff
@@ -489,11 +491,134 @@ func (h *Handler) constructStorageCluster(ds *appsv1.DaemonSet) (*corev1.Storage
 		return nil, err
 	}
 
-	// TODO: Handle kvdb secret
+	// Handle external kvdb secret migration if configured
+	if err := h.configureExternalKvdbAuth(cluster, ds, kvdbSecret.ca, kvdbSecret.cert, kvdbSecret.key, kvdbSecret.aclToken, kvdbSecret.userpwd); err != nil {
+		return nil, err
+	}
 
 	return cluster, nil
 }
 
+// configureExternalKvdbAuth migrate external kvdb auth configurations
+// Certification auth:
+// 1. Reuse px-kvdb-auth secret if cert file names are operator default ones
+// 2. Create a new secret if cert file names are customized
+// ACL token:
+// 1. Reuse px-kvdb-auth secret if it has key acl-token
+// 2. Create a new secret and store the arg value to it
+// Password auth:
+// 1. Reuse px-kvdb-auth secret if it has both key username and password
+// 2. Create a new secret and store the arg values to it
+func (h *Handler) configureExternalKvdbAuth(cluster *corev1.StorageCluster, ds *appsv1.DaemonSet, caPath, certPath, keyPath, aclToken, userpwd string) error {
+	useCertificate := caPath != "" || certPath != "" || keyPath != ""
+	useACLToken := aclToken != ""
+	usePassword := userpwd != ""
+	if !useCertificate && !useACLToken && !usePassword {
+		return nil
+	}
+
+	// Need to create a new secret for cert auth if the keys don't match
+	createNewSecret := false
+	if caPath != "" && path.Base(caPath) != portworx.SecretKeyKvdbCA ||
+		certPath != "" && path.Base(certPath) != portworx.SecretKeyKvdbCert ||
+		keyPath != "" && path.Base(keyPath) != portworx.SecretKeyKvdbCertKey {
+		createNewSecret = true
+	}
+
+	// Need to create a new secret for acl-token and password auth if secret or key not found
+	secret := &v1.Secret{}
+	secretName := "px-kvdb-auth"
```

Comment:
Let's not assume the name of the secret. We can get the name of the secret from the daemonset volumes list, right?
If we pick the secret name from the volumes, then we won't have to throw the error on line 563 ever.